components/TemplateGrid/TemplateGridContainer.vue

```diff
@@ -45,202 +45,142 @@ export default {
-            const ghostElementRow = this.ghostElement ? this.ghostElement.row : null;
-            if (
-                this.ghostElementRow !== null
-                && row === ghostElementRow + this.positionBeetwenRows
-            ) return null;
-            const newRow = ghostElementRow !== null && row > ghostElementRow
-                ? row - 1
-                : row;
-            return this.isRowSet(newRow);
+        addItem(item) {
+            this.$emit('addItem', item);
         },
-        setAllowedColumn(column, interactionItem, relation = false) {
-            const { column: interactionItemColumn, row: interactionItemRow } = interactionItem;
-            const topBrother = this.isRowSet(interactionItemRow - 1);
-            if (column === null) return null;
-            if (relation) {
-                if (topBrother && column > topBrother.column + 1 && relation === 'brother') return null;
-                return column > interactionItemColumn ? interactionItemColumn + 1 : column;
+        getBottomCollidingColumn({ adjacentElColumn, collidingElColumn }) {
+            const { overColumn } = this.mousePosition;
+            if (adjacentElColumn !== null && collidingElColumn < adjacentElColumn) {
+                return adjacentElColumn;
             }
-            return column > interactionItemColumn ? interactionItemColumn + 1 : column;
+            return overColumn > collidingElColumn ? collidingElColumn + 1 : collidingElColumn;
         },
-        setAllowedRow(interactionRow, relation) {
-            return relation === 'brother' ? interactionRow - 1 : interactionRow;
-        },
-        hasChildren(id) {
-            return this.dataWithoutGhostElement.some(element => element.parent === id);
-        },
-        setCollapseParent(id) {
-            return Object.keys(this.hiddenItems).reduce((acc, el) => {
-                let x = acc;
-                if (this.hiddenItems[el].some(e => e === id)) {
-                    x = el;
-                }
-                return x;
-            }, '');
+        getTopCollidingColumn({ adjacentElColumn, collidingElColumn }) {
+            const { overColumn } = this.mousePosition;
+            if (overColumn >= collidingElColumn && overColumn <= adjacentElColumn + 1) {
+                return overColumn;
+            }
+            return collidingElColumn;
         },
-        setRowIfCollapse(row) {
-            const item = this.isRowSet(row);
-            if (!item) return null;
-            const hiddenItemParent = this.setCollapseParent(item.id);
-            if (!hiddenItemParent) return null;
-            const collapseItem = this.hiddenItems[hiddenItemParent];
-            const lastCollapseChildId = collapseItem[collapseItem.length - 1];
-            const [lastItem] = this.dataWithoutGhostElement.filter(
-                el => el.id === lastCollapseChildId,
+        getCollidingPosition(collidingEl) {
+            const { collisionRelation } = this.mousePosition;
```

Comment:
collisionRelation doesnt match to the context of mousePosition object 'mousePosition'